src/clients/BundleDataClient/utils/FillUtils.ts

```diff
@@ -75,12 +66,26 @@ export async function verifyFillRepayment(
   _fill: FillWithBlock,
   destinationChainProvider: providers.Provider,
   matchedDeposit: DepositWithBlock,
-  possibleRepaymentChainIds: number[] = []
+  hubPoolClient: HubPoolClient
 ): Promise<FillWithBlock | undefined> {
   const fill = _.cloneDeep(_fill);
 
   const repaymentChainId = getRepaymentChainId(fill, matchedDeposit);
-  const validEvmRepayment = isEvmRepaymentValid(fill, repaymentChainId, possibleRepaymentChainIds);
+  const validEvmRepayment = isEvmRepaymentValid(fill, repaymentChainId);
+  if (!isSlowFill(fill)) {
+    try {
+      const l1TokenCounterpart = hubPoolClient.getL1TokenForL2TokenAtBlock(
+        fill.inputToken,
+        fill.originChainId,
+        matchedDeposit.quoteBlockNumber
+      );
+      hubPoolClient.getL2TokenForL1TokenAtBlock(l1TokenCounterpart, repaymentChainId, matchedDeposit.quoteBlockNumber);
+      // Repayment token could be found, this is a valid repayment chain.
+    } catch {
```

Comment:
This universal try catch is quite dangerous. Should we fix this now or in a follow up.

One way to mostly fix it is to make the hub error a specific type and catch only that werror type.

Reply:
changed behavior to switch to destination chain

Reply:
Still dangerous as a bug will silently change repayment location, causing invalid bundles.

We can resolve in follow up.

Reply:
but wouldn't that indicate the bug that causes the invalid bundles is an invalid code change?

Reply:
We want an error in an inner function to bubble up, not silently create bad bundles.

My point is that we're assuming the error is X when we could be catching error Y without realizing it.

Reply:
> This universal try catch is quite dangerous. Should we fix this now or in a follow up?
> 
> One way to mostly fix it is to make the hub error a specific type and catch only that werror type.

We started on this a while ago and should pick it up again: https://github.com/across-protocol/sdk/pull/643